cmd/zpool/zpool_vdev.c

```diff
@@ -829,7 +498,7 @@ get_replication(nvlist_t *nvroot, boolean_t fatal)
 			rep.zprl_children = 1;
 			rep.zprl_parity = 0;
 		} else {
-			uint64_t vdev_size;
+			int64_t vdev_size;
```

Comment:
Could you explain the reason behind all these "unit64_t" changes? Doesn't seem Linux specific functionality.

Reply:
`labs(size - vdev_size)` triggers a clang warning "taking the absolute value of unsigned type 'unsigned long' has no effect"

Reply:
> 
> 
> `labs(size - vdev_size)` triggers a clang warning "taking the absolute value of unsigned type 'unsigned long' has no effect"

Then the right solution might be to drop the labs() rather than change the type. The type should definitely stay uint64_t

Reply:
> > `labs(size - vdev_size)` triggers a clang warning "taking the absolute value of unsigned type 'unsigned long' has no effect"
> 
> Then the right solution might be to drop the labs() rather than change the type. The type should definitely stay uint64_t

Not if size can be less than vdev_size.

Reply:
`size` and `vdev_size` will always be representable by 63-bits so keeping it unsigned is gratuitous if we aim to take the difference of the two.

Reply:
The difference between the sizes could be positive or negative. Leaving the types as unsigned means the result overflows when the difference is negative and removing the labs() means we'll have introduced a bug. The subtraction results in the correct value when the unsigned integer is interpreted as a signed integer by labs().

Clang doesn't see that we're doing a subtraction and abusing the types. It sees the result of the subtraction, an unsigned value, being passed to an absolute value function and emits a warning which we treat as an error.

Reply:
We've ended up making similar fixes throughout the code for issues like this; e.g. when it's expected that the value can may be negative.  But since this change is independent of the proposed refactoring, let's open a new PR with only this fix and a brief explanation.  @freqlabs's comment above would be great.

Reply:
#9335